rfc/rfc-100/rfc-100.md

```diff
@@ -0,0 +1,238 @@
+- **Unstructured Column Groups**: Use Lance format for vector/embedding data or specially configured Parquet for BLOB storage
+
+**Format Configuration**: File format is determined at column group creation time based on (per the current RFC-80). 
+But, ideally all these configurations should be automatic and Hudi should auto-generate colum group names and mappings.
+
+
+```sql
+CREATE TABLE multimedia_catalog (
+  id BIGINT,
+  product_name STRING,
+  category STRING,
+  image BINARY,
+  video LARGE_BINARY,
+  embeddings ARRAY<FLOAT>
+) USING HUDI
+TBLPROPERTIES (
+  'hoodie.table.type' = 'MERGE_ON_READ',
+  'hoodie.bucket.index.hash.field' = 'id',
+  'hoodie.columngroup.structured' = 'id,product_name,category;id',
+  'hoodie.columngroup.images' = 'id,image;id',
+  'hoodie.columngroup.videos' = 'id,video;id',
+  'hoodie.columngroup.ml' = 'id,embeddings;id',
+  'hoodie.columngroup.images.format' = 'parquet',
+  'hoodie.columngroup.videos.format' = 'lance',
+  'hoodie.columngroup.ml.format' = 'hfile'
+)
+```
+
+### 2. Dynamic Inline/Out-of-Line Storage
+
+**Storage Decision Logic**: During write time, Hudi determines storage strategy based on:
+- **Inline Storage**: BLOB data < 1MB stored directly in the column group file, to avoid excessive cloud storage API calls.
+- **Out-of-Line Storage**: Large BLOB data stored in dedicated object store locations with pointers in the main file, to avoid write amplification during updates.
+
+
+**Storage Pointer Schema**:
+```json
+{
+  "type": "record",
+  "name": "BlobPointer",
+  "fields": [
+    {"name": "storage_type", "type": "string"},
+    {"name": "size", "type": "long"},
+    {"name": "checksum", "type": "string"},
+    {"name": "external_path", "type": ["null", "string"]},
+    {"name": "compression", "type": ["null", "string"]}
+  ]
+}
+```
+
+**External Storage Layout**:
+```
+{table_path}/.hoodie/blobs/{partition}/{file_group_id}/{column_group}/{instant}/{blob_id}
+```
+Alternatively, User should be able to specify external storage location per BLOB during writes, as needed.
+
+### 3. Parquet Optimization for BLOB Storage
+
+For unstructured column groups using Parquet:
+- **Disable Compression**: Avoid double compression of already compressed media files
+- **Plain Encoding**: Use PLAIN encoding instead of dictionary encoding for BLOB columns
+- **Large Page Sizes**: Configure larger page sizes to optimize for sequential BLOB access
+- **Metadata Index**: Maintain BLOB metadata in Hudi metadata table for efficient retrieval of a single blob value.
+- **Disable stats**: Not very useful for BLOB columns
+
+### 4. Lance Format Integration
+
+**Lance Advantages for Unstructured Data**:
+- Native support for high-dimensional vectors and embeddings
+- Efficient columnar storage for mixed structured/unstructured data
+- Better compression for certain unstructured data types
+
+Supporting Lance, working across Hudi + Lance communities will help users unlock benefits of both currently supported 
```

Comment:
Curious on what comparisions were done between different formats in this space? 

is our long term goal to have this unstructured file format be pluggable similar to what what we have do already for hudi on structured side with parquet and orc etc https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFileFormat.java#L34